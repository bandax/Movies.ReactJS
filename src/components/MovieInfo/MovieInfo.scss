.movie-info-window {
  background-color: #000000;
  height: 400px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

.movie-info-modal {
  padding: 20px;
  background: #000000;
  border-radius: 5px;
  transition: all 5s ease-in-out;
}

.movie-image {
  text-align: center;
}
