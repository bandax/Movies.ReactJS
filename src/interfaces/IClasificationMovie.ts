export interface IClasification {
    id: string;
    name: string;
}
