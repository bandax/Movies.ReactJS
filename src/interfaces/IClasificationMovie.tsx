export interface IClasification {
    id: string,
    name: string
}

export interface IClasificationMovieProps {
    clasificationTypes: IClasification[],    
}

