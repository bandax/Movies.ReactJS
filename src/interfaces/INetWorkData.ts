export interface INetworkData {
  url: string;
  body: string;
  contentType: string;
}
