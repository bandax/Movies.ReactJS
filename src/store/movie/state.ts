import { IMovie } from "../../interfaces/IResultsMovies";

export interface MovieState {
  movies: IMovie[];
  loading: boolean;
  errorMessage: string;
}
