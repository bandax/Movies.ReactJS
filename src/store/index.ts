import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { movieReducer } from './movie/reducers';

const rootReducer = combineReducers({
  movie: movieReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(rootReducer, applyMiddleware(thunk));
