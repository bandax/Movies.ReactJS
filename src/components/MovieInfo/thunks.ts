import { Action } from 'redux';
import {
  loadingMovies,
  loadingMoviesSuccess,
  loadingMoviesError,
  selectMovie,
} from '../../store/movie/actions';
import { RootState } from '../../store/index';
import { ThunkDispatch } from 'redux-thunk';
import { getRequest } from '../../services/networkService';
import { IMovieData } from '../../interfaces/IMovieData';

export const getMovieById = (id: string) => async (
  dispatch: ThunkDispatch<RootState, void, Action>
) => {
  try {
    dispatch(loadingMovies());
    const response = await getRequest(`http://localhost:4000/movies/${id}`);
    const movie: IMovieData = await response.json();
    dispatch(selectMovie(movie));
  } catch (error) {
    dispatch(loadingMoviesError(error.message));
  }
};
