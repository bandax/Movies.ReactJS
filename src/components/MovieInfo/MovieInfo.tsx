import * as React from 'react';
import './MovieInfo.scss';
import { IMovie } from '../../interfaces/IResultsMovies';
import { useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import { RootState } from '../../store/index';
import { getMovieById } from './thunks';
import { ThunkDispatch } from 'redux-thunk';
import { getMoviesLoading, getMoviesData } from '../ResultsMovie/selectors';
import { Action } from 'redux';

interface IParamTypes {
  movieId: string;
}

const mapStateToProps = (state: RootState) => ({
  movie: state.movie.movie,
  movies: getMoviesData(state),
  isLoading: getMoviesLoading(state),
});

const mapDispatchToProps = (
  dispatch: ThunkDispatch<RootState, void, Action>
) => {
  return {
    getMovieById: (id: string) => dispatch(getMovieById(id)),
  };
};

type Props = ReturnType<typeof mapStateToProps> &
  ReturnType<typeof mapDispatchToProps>;

const MovieInfo: React.FunctionComponent<Props> = ({
  movie,
  movies,
  isLoading,
  getMovieById,
}) => {
  const { movieId } = useParams<IParamTypes>();

  React.useEffect(() => {
    getMovieById(movieId);
  }, []);

  const getYear = function (releaseDate: string) {
    const date = new Date(releaseDate);
    return date.getFullYear();
  };

  const loadingMessage = <div>Searching movies...</div>;
  const content = (
    <>
      <div className="row">
        <div className="col-4 movie-image">
          <img width="50%" className="poster-movie" src={movie?.poster_path} />
        </div>
        <div className="col-8">
          <div className="">
            <h1 className="movie-title" data-testid="movie-title">
              {movie?.title}
            </h1>{' '}
            Rate: <span className="movie-rate">{movie?.vote_average}</span>
          </div>
          <div>
            Release Date:{' '}
            <span className="movie-year">{getYear(movie?.release_date)}</span>{' '}
          </div>
          <div>
            Overview: <p className="movie-review">{movie?.overview}</p>
          </div>
        </div>
      </div>
    </>
  );

  return isLoading ? loadingMessage : content;
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieInfo);
