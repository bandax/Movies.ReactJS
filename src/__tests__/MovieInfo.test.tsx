import * as React from 'react';
import MovieInfo from '../components/MovieInfo/MovieInfo';
import thunk, { ThunkDispatch } from 'redux-thunk';
import {
  render,
  fireEvent,
  waitFor,
  screen,
  getByText,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import clasificationTypes from '../data/clasifications.json';
import { IMovieData } from '../interfaces/IMovieData';
import {
  HashRouter as Router,
  Route,
  Switch,
  MemoryRouter,
} from 'react-router-dom';

import configureMockStore, { MockStoreCreator } from 'redux-mock-store';
import { MovieState } from '../store/movie/state';
import { Action } from 'redux';
import { Provider } from 'react-redux';
import { RootState } from '../store/index';

const movie: IMovieData = {
  id: 1234,
  title: 'Avengers Infinity War',
  poster_path:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png',
  release_date: new Date().toLocaleDateString(),
  budget: 123,
  revenue: 123,
  vote_average: 123,
  vote_count: 123,
  runtime: 412,
  tagline: '12343',
  overview: 'tests',
  genres: ['Action & Adventure', 'Comedy'],
};

type DispatchExts = ThunkDispatch<MovieState, void, Action>;
const middleware = [thunk];
const state: RootState = {
  movie: {
    movie: movie,
    errorMessage: '',
    findMovies: true,
    loading: false,
    movies: [],
  },
};

const mockStore = configureMockStore<RootState, DispatchExts>(middleware);

describe('<MovieInfo />', () => {
  let onAddMovieSubmit;
  let onShowAddMovieWindow;

  let store;

  beforeEach(() => {
    store = mockStore(state);
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
      useParams: () => ({
        movieId: 1234,
      }),
    }));
  });

  //   beforeEach(() => {
  //     onAddMovieSubmit = jest.fn();
  //     onShowAddMovieWindow = jest.fn();
  //   });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render a MovieInfo Component', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/film/1234']}>
          <Route path="/film/:movieId">
            <MovieInfo />
          </Route>
        </MemoryRouter>
      </Provider>
    );

    const fragment = asFragment();

    expect(fragment).toMatchSnapshot();
  });

  it('should render movie data', async () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/film/1234']}>
          <Route path="/film/:movieId">
            <MovieInfo />
          </Route>
        </MemoryRouter>
      </Provider>
    );

    const title = getByTestId('movie-title');
    screen.debug();

    expect(title).toBeInTheDocument();
  });
});
