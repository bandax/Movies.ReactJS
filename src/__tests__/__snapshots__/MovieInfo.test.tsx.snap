// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieInfo /> should render a MovieInfo Component 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-4 movie-image"
    >
      <img
        class="poster-movie"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png"
        width="50%"
      />
    </div>
    <div
      class="col-8"
    >
      <div
        class=""
      >
        <h1
          class="movie-title"
          data-testid="movie-title"
        >
          Avengers Infinity War
        </h1>
         Rate: 
        <span
          class="movie-rate"
        >
          123
        </span>
      </div>
      <div>
        Release Date: 
        <span
          class="movie-year"
        >
          2020
        </span>
         
      </div>
      <div>
        Overview: 
        <p
          class="movie-review"
        >
          tests
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
